rest/access/handlers.go

```diff
@@ -53,9 +54,18 @@ func Authenticate(state *state.State, r *http.Request, hostAddress string, trust
 		return true, nil
 	}
 
-	if state.Address().URL.Host == "" {
-		logger.Info("Allowing unauthenticated request to un-initialized system")
-		return true, nil
+	// Check if the listener is using the server.crt.
+	// This indicates that the daemon is in a pre-init state and is listening on the PreInitListenAddress.
+	serverCertificate := state.ServerCert()
+	networkEndpoints := state.Endpoints.List(endpoints.EndpointNetwork)
```

Comment:
Hmm, I wonder if it's safe to drop all authentication if _any_ listener uses the `server.crt`. I think we should only check the `core` listener. 

It's a small edge case, but if you have an extension server that perpetually uses the `server.crt`, then every request to every endpoint would always skip authentication.

Reply:
True, if somebody reads the `server` keypair and pushes it to the `/certificates` dir this will have the side effect that every request can run unauthenticated.

Fixed!